lib/msal-electron-proof-of-concept/src/AppConfig/ClientApplication/PublicClientApplication.ts

```diff
@@ -0,0 +1,161 @@
+ *
+ * This class can be instantiated into objects that the developer
+ * can use in order to acquire tokens.
+ */
+export class PublicClientApplication extends ClientApplication {
+    private authWindow: BrowserWindow;
+    private authCodeListener: AuthCodeListener;
+
+    constructor(authOptions: AuthOptions) {
+        super(authOptions);
+    }
+
+    /**
+     * The acquireToken async method uses the Authorization Code
+     * Grant to retrieve an access token from the AAD authorization server,
+     * which can be used to make authenticated calls to an resource server
+     * such as MS Graph.
+     */
+    public async acquireToken(request: AuthenticationParameters): Promise<string> {
+        return new Promise<string>((resolve, reject) => {
+            try {
+                // Validate and filter scopes
+                this.validateInputScopes(request.scopes);
+                // Set Authority URL from developer input or default if not in request
+                const authorityUrl = request.authority ? request.authority : this.authorityUrl;
+                // Create Authority Instance
+                const authorityInstance = new AadAuthority(authorityUrl);
+                // Get Authorization Code
+                const authCode = this.retrieveAuthCode(authorityInstance, request.scopes);
+                resolve(authCode);
+            } catch (error) {
+                return reject(error);
+            }
+        });
+    }
+
+    /**
+     * Executes a series of checks and validations on the scopes array
+     * that was sent in the PublicClientApplication's constructor.
+     * @param scopes
+     */
+    private validateInputScopes(scopes: string[]): void {
+        // Throws if scopes object is not truthy
+        assert(scopes, ClientConfigurationError.createScopesRequiredError(scopes));
+
+        // Throws if scopes object is not an array.
+        if (!Array.isArray(scopes)) {
+            throw ClientConfigurationError.createScopesNonArrayError(scopes);
+        }
+
+        // Throws if scopes array is empty.
+        if (scopes.length < 1) {
+            throw ClientConfigurationError.createEmptyScopesArrayError(scopes);
+        }
+    }
+
+    /**
+     * This method is responsible for requesting and returning an authorization code
+     * from the authorization endpoint of the authorization server.
+     * @param authorityInstance
+     * @param scopes
+     */
+    private async retrieveAuthCode(authorityInstance: Authority, scopes: string[]): Promise<string> {
+        // Register custom protocol to listen for auth code response
+        this.authCodeListener = new CustomFileProtocolListener('msal');
+        this.authCodeListener.start();
+
+        // Build navigate URL for Auth Code request
+        const navigateUrl = this.buildAuthCodeUrl(authorityInstance, scopes);
+
+        try {
+            return await this.listenForAuthCode(navigateUrl);
+        } catch (error) {
+            throw error;
```

Comment:
Don't need to wrap in ` try/catch` if you are just going to rethrow the error.

Reply:
Makes sense, I had an UnhandledPromiseRejection error and forgot to remove this try/catch block after debugging.